### PR TITLE
Add unit test for pkg/lint/rules/deprecations.go

### PR DESCRIPTION
**Pull request by hs0210**
Signed-off-by: Hu Shuai <hus.fnst@cn.fujitsu.com>

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by hickeyma — CHANGES_REQUESTED** [review]
Thanks for pushing the PR @hs0210. I am all for tests but I am unsure what this test adds. Can you explain what this test provides that is not covered by `TestValidateNoDeprecations`?

**Comment by hs0210** [comment]
Thanks for your review @hickeyma . Though the fmt.Sprintf() has been tested, the whole logic is worth a test. And the coverage is improved.

**Review by jdolitsky — APPROVED** [review]

File: pkg/lint/rules/deprecations_test.go [hunk 1]
```diff
@@ -40,3 +40,11 @@ func TestValidateNoDeprecations(t *testing.T) {
 		t.Errorf("Expected a v1 Pod to not be deprecated")
 	}
 }
+
+func TestError(t *testing.T) {
```
Comment (technosophos) [inline pkg/lint/rules/deprecations_test.go:44]: This name is ambiguous and should be replaced by something like `TestDeprecatedAPIError`

**Review by technosophos — CHANGES_REQUESTED** [review]

**Comment by hs0210** [comment]
@technosophos PTAL.

**Pull request closed without merging**